shell/platform/darwin/ios/framework/Source/FlutterPlatformViews.mm

```diff
@@ -614,6 +656,22 @@ - (BOOL)gestureRecognizer:(UIGestureRecognizer*)gestureRecognizer
   return otherGestureRecognizer == self;
 }
 
+- (void)touchesBegan:(NSSet<UITouch*>*)touches withEvent:(UIEvent*)event {
+  self.touchedEndedWithoutBlocking = NO;
+  [super touchesBegan:touches withEvent:event];
+}
+
+- (void)touchesEnded:(NSSet<UITouch*>*)touches withEvent:(UIEvent*)event {
+  if (self.shouldEndInNextTouchesEnded) {
+    NSLog(@"block gesture in touches ended");
+    self.state = UIGestureRecognizerStateEnded;
+    self.shouldEndInNextTouchesEnded = NO;
+    return;
+  }
+  self.touchedEndedWithoutBlocking = YES;
```

Comment:
Do we need to set this here?

Reply:
If `releaseGesture` is called after `touchesEnded`(very rarely), we need to do `self.state = UIGestureRecognizerStateEnded;` the in `releaseGesture` method, since the touchesEnded won't be called again. Setting this kinda let the `releaseGesture` know touchesEnded has happened already.